pkg/sql/plan/build_ddl.go

```diff
@@ -385,13 +385,37 @@ func buildTableDefs(defs tree.TableDefs, ctx CompilerContext, tableDef *TableDef
 	}
 
 	if len(primaryKeys) > 0 {
-		tableDef.Defs = append(tableDef.Defs, &plan.TableDef_DefType{
-			Def: &plan.TableDef_DefType_Pk{
-				Pk: &plan.PrimaryKeyDef{
-					Names: primaryKeys,
+		if len(primaryKeys) == 1 {
+			tableDef.Defs = append(tableDef.Defs, &plan.TableDef_DefType{
+				Def: &plan.TableDef_DefType_Pk{
+					Pk: &plan.PrimaryKeyDef{
+						Names: primaryKeys,
+					},
 				},
-			},
-		})
+			})
+		} else {
+			cPkeyName := util.BuildCompositePrimaryKeyColumnName(primaryKeys)
+			tableDef.Cols = append(tableDef.Cols, &ColDef{
+				Name: cPkeyName,
+				Alg:  plan.CompressType_Lz4,
+				Typ: &Type{
+					Id:   int32(types.T_varchar),
+					Size: types.VarlenaSize,
+				},
+				Default: &plan.Default{
+					NullAbility:  false,
+					Expr:         nil,
+					OriginString: "",
+				},
+			})
+			tableDef.Defs = append(tableDef.Defs, &plan.TableDef_DefType{
```

Comment:
This won't work in general, what if user alter table rename a column?    Unless you take care of that fixing table defs.
Do we?

Reply:
I think alter is not implement yet, when alter is implemented, I need to add some code